code/modules/bitrunning/components/glitch.dm

```diff
@@ -0,0 +1,89 @@
+/datum/component/glitch
+	/// Ref of the spawning forge
+	var/datum/weakref/forge_ref
+
+/datum/component/glitch/Initialize(obj/machinery/quantum_server/server, obj/machinery/byteforge/forge)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(forge, COMSIG_MACHINERY_POWER_RESTORED, PROC_REF(on_forge_power_restored))
+	RegisterSignals(forge, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_forge_broken))
+	forge_ref = WEAKREF(forge)
+
+	var/mob/living/owner = parent
+	server.spawned_threat_refs.Remove(WEAKREF(owner)) // so the server doesn't dust us
+
+	owner.faction.Cut()
+	owner.faction += list(ROLE_GLITCH)
+
+	var/current_max = owner.maxHealth + ROUND_UP(server.threat * 0.2)
+	owner.maxHealth = clamp(current_max, 200, 500)
+	owner.fully_heal()
+
+	owner.AddElement(/datum/element/digital_aura)
+
+/datum/component/glitch/RegisterWithParent()
+	RegisterSignals(parent, list(COMSIG_LIVING_STATUS_UNCONSCIOUS, COMSIG_LIVING_DEATH), PROC_REF(on_death))
+
+/datum/component/glitch/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_LIVING_DEATH, PROC_REF(on_death))
+
+/// Sakujo
+/datum/component/glitch/proc/dust_mob()
+	if(QDELETED(parent))
+		return
+
+	var/mob/living/owner = parent
+	owner.dust()
+	qdel(src)
+
+/// We don't want digital entities just lingering around as corpses.
+/datum/component/glitch/proc/on_death()
+	SIGNAL_HANDLER
+
+	if(QDELETED(parent))
+		return
+
+	var/mob/living/owner = parent
+	to_chat(owner, span_userdanger("You feel a strange sensation..."))
+
+	var/obj/machinery/byteforge/forge = forge_ref.resolve()
+	forge?.setup_particles()
+
+	addtimer(CALLBACK(src, PROC_REF(dust_mob)), 2 SECONDS, TIMER_UNIQUE|TIMER_DELETE_ME|TIMER_STOPPABLE)
+
+/// If the forge breaks, we take a massive slowdown
+/datum/component/glitch/proc/on_forge_broken(datum/source)
+	SIGNAL_HANDLER
+
+	var/mob/living/player = parent
+	var/atom/movable/screen/alert/bitrunning/alert = player.throw_alert(
+		ALERT_BITRUNNER_GLITCH,
+		/atom/movable/screen/alert/bitrunning,
+		new_master = source,
+	)
+	alert.name = "Source Broken"
+	alert.desc = "Our byteforge has been broken."
```

Comment:
subtype?

Reply:
I've been on a kick of just reusing the same alert - they can return to subtypes? 

(Later I'd like to just make name/desc an argument on throw_alert)

Reply:
ah right, nvm then